azurerm/resource_arm_api_management.go

```diff
@@ -0,0 +1,773 @@
+			return fmt.Errorf("Error setting `security`: %+v", err)
+		}
+
+		if err := d.Set("hostname_configuration", flattenApiManagementHostnameConfigurations(d, props.HostnameConfigurations)); err != nil {
+			return fmt.Errorf("Error setting `hostname_configuration`: %+v", err)
+		}
+
+		if err := d.Set("additional_location", flattenApiManagementAdditionalLocations(props.AdditionalLocations)); err != nil {
+			return fmt.Errorf("Error setting `additional_location`: %+v", err)
+		}
+
+		if err := d.Set("certificate", flattenApiManagementCertificates(d, props.Certificates)); err != nil {
+			return fmt.Errorf("Error setting `certificate`: %+v", err)
+		}
+	}
+
+	if sku := resp.Sku; sku != nil {
+		if err := d.Set("sku", flattenApiManagementServiceSku(sku)); err != nil {
+			return fmt.Errorf("Error flattening `sku`: %+v", err)
+		}
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func flattenApiManagementCustomProperties(input map[string]*string) ([]interface{}, error) {
+	output := make(map[string]interface{}, 0)
+
+	if v := input["Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Ssl30"]; v != nil {
+		val, err := strconv.ParseBool(*v)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing `disable_backend_ssl30` %q: %+v", *v, err)
+		}
+		output["disable_backend_ssl30"] = val
+	}
+
+	if v := input["Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls10"]; v != nil {
+		val, err := strconv.ParseBool(*v)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing `disable_backend_tls10` %q: %+v", *v, err)
+		}
+		output["disable_backend_tls10"] = val
+	}
+
+	if v := input["Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Backend.Protocols.Tls11"]; v != nil {
+		val, err := strconv.ParseBool(*v)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing `disable_backend_tls11` %q: %+v", *v, err)
+		}
+		output["disable_backend_tls11"] = val
+	}
+
+	if v := input["Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Ciphers.TripleDes168"]; v != nil {
+		val, err := strconv.ParseBool(*v)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing `disable_triple_des_chipers` %q: %+v", *v, err)
+		}
+		output["disable_triple_des_chipers"] = val
+	}
+
+	if v := input["Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Ssl30"]; v != nil {
+		val, err := strconv.ParseBool(*v)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing `disable_frontend_ssl30` %q: %+v", *v, err)
+		}
+		output["disable_frontend_ssl30"] = val
+	}
+
+	if v := input["Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls10"]; v != nil {
+		val, err := strconv.ParseBool(*v)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing `disable_frontend_tls10` %q: %+v", *v, err)
+		}
+		output["disable_frontend_tls10"] = val
+	}
+
+	if v := input["Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls11"]; v != nil {
```

Comment:
Could this changed to use a local helper?

```go
if err := setCustomPropertyFrom(input, "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls11", output, "disable_frontend_tls11"); err != nil {
    return nil, err
}

setCustomPropertyFrom(input, path, output, key) {
  if v := input[path]; v != nil {
    val, err := strconv.ParseBool(*v)
    if err != nil {
      return nil, fmt.Errorf("Error parsing `%s` %q: %+v", key, *v, err)
    }
    output[key] = val
  }

  return nil 
}
```

not blocking the pr, just a idea